message.module

```diff
@@ -108,127 +108,11 @@ function message_entity_delete(EntityInterface $entity) {
  * Fetch all message templates and purge old messages.
  */
 function message_cron() {
-  $config = \Drupal::config('message.settings');
-
-  // The maximal amount of messages to purge per cron run.
-  $purge_limit = $config->get('delete_cron_limit');
-
-  // Messages to be deleted.
-  $purge_messages = [];
-  // Names of non global-purge-settings overriding message templates.
-  $no_override_template_names = [];
-  // Message templates that override global purge settings.
-  $override_templates = [];
-
-  // Iterate all message templates to distinguish between overriding and non-
-  // overriding templates.
-  /** @var MessageTemplate[] $message_templates */
-  $message_templates = MessageTemplate::loadMultiple();
-  foreach ($message_templates as $message_template) {
-    $settings = $message_template->getSetting('purge');
-    if (empty($settings['override'])) {
-      $no_override_template_names[] = $message_template->id();
-    }
-    else {
-      // For overriding templates, store the template and not its name to later extract
-      // the specific purge settings.
-      $override_templates[] = $message_template;
-    }
-  }
-
-  // Gather purgeable messages of overriding templates.
-  /** @var MessageTemplate[] $override_templates */
-  foreach ($override_templates as $message_template) {
-    // Ignore message template with unchecked "Purge messages". Also make sure that
-    // purging settings are present.
-    $settings = $message_template->getSetting('purge');
-    if (empty($settings['enabled'])) {
-      continue;
-    }
-
-    $purge_messages += message_get_purgeable_by_template($purge_limit, $message_template->id(), $settings);
-  }
-
-  // Gather purgeable messages of non-overriding templates according to global
-  // settings.
-  if (!empty($no_override_template_names)) {
-    // Do nothing if purge isn't enabled.
-    if ($config->get('purge_enable', FALSE)) {
-      $purge_settings = [
-        'quota' => $config->get('purge_quota'),
-        'days' => $config->get('purge_days'),
-      ];
-
-      $purge_messages += message_get_purgeable_by_template($purge_limit, $no_override_template_names, $purge_settings);
-    }
-  }
-
-  // Delete all gathered messages.
-  if (!empty($purge_messages)) {
-    \Drupal::entityManager()->getStorage('message')->delete($purge_messages);
-  }
+  /** @var \Drupal\message\MessagePurger $message_purger */
+  $message_purger = \Drupal::service('message.purger');
```

Comment:
maybe we should call it something like `$message_purge_orchestrator`, just so the distinction is clearer between this and the `MessagePurge` plugins?